models/Asset.php

```diff
@@ -1108,6 +1111,21 @@ public function getStream()
         return $this->stream;
     }
 
+    public function getChecksum(): string
+    {
+        if (!$checksum = $this->getCustomSetting('checksum')) {
+            $this->generateChecksum();
+            $checksum = $this->getCustomSetting('checksum');
+        }
+
+        return $checksum;
+    }
+
+    public function generateChecksum(): void
+    {
+        $this->setCustomSetting('checksum', Storage::get('asset')->checksum($this->getRealFullPath()));
```

Comment:
```suggestion
        $this->setCustomSetting('checksum', Storage::get('asset')->checksum($this->getRealFullPath()));
        $this->getDao()->updateCustomSettings();
```
We could persist the info right away, right?

Reply:
yes, done.